### PR TITLE
Updates to maven migration guide

### PR DESCRIPTION
**Pull request by ljacomet**
* Remove reference to luck
* Add info about using java-platform for sharing versions
* Misc improvements

### CODE REVIEW

File: subprojects/docs/src/docs/userguide/migrating_from_maven.adoc [hunk 1]
```diff
@@ -172,7 +172,7 @@ Options include:
  * <<working_with_files#sec:creating_archives_example,Creating custom archive tasks>>
  * Using a suitable community plugin from the https://plugins.gradle.org/[Gradle Plugin Portal]
 
-If you're lucky and don't have many plugins or much in the way of customisation in your Maven build, you can simply run
+If you don't have many plugins or much in the way of customisation in your Maven build, you can simply run
```
Comment (jjohannes) [inline subprojects/docs/src/docs/userguide/migrating_from_maven.adoc]: Suggestion:

_If your maven build follows standard conventions and does not apply any specific plugins, you can simply run_

**Comment by ljacomet** [comment]
Of course, I saw it and wanted to add a TODO. Thanks for catching it

File: subprojects/docs/src/docs/userguide/migrating_from_maven.adoc (hunk 1 above)
Comment (ljacomet) [reply subprojects/docs/src/docs/userguide/migrating_from_maven.adoc]: Adapted but not 100% like suggested. After all plugins like surefire and others are not an issue at all.

**Comment by ljacomet** [comment]
@jjohannes Updated following your review, please have another look

**Pull request merged**